app.js

```diff
@@ -254,6 +256,18 @@ async function downloadFile(remoteObject, headers, credentialsType) {
       //--- write the file
       try {
         await saveFileToDisk(response, localAddress);
+
+        const contentType = response.headers.get('content-type');
+
+        if (GUESS_FILE_TYPE_BINARY_FILES && contentType == 'application/octet-stream') {
+          const newExtension = await getRealExtension(localAddress);
```

Comment:
I also had a thought, couldn't we also try to deduce a potential extension from the request URL itself?

Reply:
I'll work on the readability.

For the part about deducing if a file is html, I think we have two cases here
1. Our file doesn't have a content type -> default it to html (because of some convention with the users I suppose)
2. Our file has a content type -> we can try to deduce the extension from that. But when it's `application/octet-stream`, which is always the case when downloading a file from a sharepoint apparently, the extension would be set to .bin even though the file is from a different type. In that case, we'll try to find its type. But here, imagine the .bin file is actually an executable : we wouldn't want to default that to a .html. That's why I try parsing the file, to be sure what i say is an html file really is one.

Does it make more sense ?

Reply:
As for deducing the extension from the request url, didn't look into it, going down that path would take me a time I don't have if we're going to release it at the end of the sprint :sweat_smile:

Reply:
Yeah, I just find a bit weird behavior. Give it no headers, it becomes a `.html`. So I can still ram an executable in there, if I wanted to. But only if I supply headers `application/octet-stream` it wil be defaulted to a `.bin` and protect us against executables. Take note, I'm saying this while not being 100% about what these executable files would look like.

Reply:
I kept the existing behaviour as it was, guessing it has been tought out at the time it got created. @cecemel do you think we should change it ?

Reply:
@claire-lovisa @HugaertsDries 
- Please no try from URL there is no standard whatsoever there

Can we not skip default html?

If content-type is not (octet-stream and empty), we use the already provided logic. Else given we want to guess it we go down that path.
--> test magic number, if it yields results we go further. Else it remains bin.
results from magic number:
if not text -> we map the extension
if text -> we want to be more specific, we test if html, else we'll keep it as text.

?